pkg/service/dwn/dwn.go

```diff
@@ -3,61 +3,69 @@ package dwn
 import (
 	"fmt"
 
+	sdkutil "github.com/TBD54566975/ssi-sdk/util"
 	"github.com/sirupsen/logrus"
 
 	"github.com/tbd54566975/ssi-service/config"
 	"github.com/tbd54566975/ssi-service/internal/util"
 	"github.com/tbd54566975/ssi-service/pkg/service/framework"
 	"github.com/tbd54566975/ssi-service/pkg/service/keystore"
-	manifeststorage "github.com/tbd54566975/ssi-service/pkg/service/manifest/storage"
+	"github.com/tbd54566975/ssi-service/pkg/service/manifest"
 	"github.com/tbd54566975/ssi-service/pkg/storage"
 )
 
 type Service struct {
-	config          config.DWNServiceConfig
-	manifestStorage manifeststorage.Storage
+	config config.DWNServiceConfig
 
 	// external dependencies
 	keyStore *keystore.Service
+	manifest *manifest.Service
 }
 
 func (s Service) Type() framework.Type {
 	return framework.DWN
 }
 
 func (s Service) Status() framework.Status {
-	if s.manifestStorage == nil {
+	err := sdkutil.NewAppendError()
+	if s.keyStore == nil {
+		err.AppendString("keystore not set")
+	}
+	if s.manifest == nil {
+		err.AppendString("manifest not set")
+	}
+	if gotErr := err.Error(); gotErr != nil {
 		return framework.Status{
 			Status:  framework.StatusNotReady,
-			Message: "no manifestStorage",
+			Message: gotErr.Error(),
 		}
 	}
-
 	return framework.Status{Status: framework.StatusReady}
 }
 
 func (s Service) Config() config.DWNServiceConfig {
 	return s.config
 }
 
-func NewDWNService(config config.DWNServiceConfig, s storage.ServiceStorage, keyStore *keystore.Service) (*Service, error) {
-	manifestStorage, err := manifeststorage.NewManifestStorage(s)
-	if err != nil {
-		errMsg := "could not instantiate manifestStorage for the dwn service"
-		return nil, util.LoggingErrorMsg(err, errMsg)
+func NewDWNService(config config.DWNServiceConfig, s storage.ServiceStorage, keyStore *keystore.Service, manifest *manifest.Service) (*Service, error) {
+	if keyStore == nil {
+		return nil, fmt.Errorf("keystore not set")
+	}
+	if manifest == nil {
+		return nil, fmt.Errorf("manifest not set")
 	}
-
 	return &Service{
-		config:          config,
-		manifestStorage: manifestStorage,
-		keyStore:        keyStore,
+		config:   config,
+		keyStore: keyStore,
+		manifest: manifest,
 	}, nil
 }
 
 func (s Service) GetManifest(request DWNPublishManifestRequest) (*DWNPublishManifestResponse, error) {
+
 	logrus.Debugf("getting manifest: %s", request.ManifestID)
 
-	gotManifest, err := s.manifestStorage.GetManifest(request.ManifestID)
+	gotManifest, err := s.manifest.GetManifest(manifest.GetManifestRequest{ID: request.ManifestID})
```

Comment:
Yup this is exactly what I was thinking about doing for a clean up 2nd pass.

We can even delete the dwn/model.go file and use CreateManifestRequest in manifest too, but that may be too much mixing